gcc warning with -Wunused-parameter

When I compile pico with my project with -Wunused-parameter I get the following warnings.  Obviously this isn't broken functionality but I thought it would be nice to remove the warning if possible.

```
picohttpparser.c:104:20: warning: unused parameter ‘buf_end’ [-Wunused-parameter]
 static const char *findchar_fast(const char *buf, const char *buf_end, const char *ranges, size_t ranges_size, int *found)
                    ^
lib/cpp/picohttpparser/picohttpparser.c:104:20: warning: unused parameter ‘ranges’ [-Wunused-parameter]
lib/cpp/picohttpparser/picohttpparser.c:104:20: warning: unused parameter ‘ranges_size’ [-Wunused-parameter]
```

--- a/picohttpparser.c
+++ b/picohttpparser.c
@@ -121,6 +121,11 @@ static const char *findchar_fast(const char *buf, const char *buf_end, const cha
             left -= 16;
         } while (likely(left != 0));
     }
+#else
+    /* suppress unused parameter warning */
+    (void)buf_end;
+    (void)ranges;
+    (void)ranges_size;
 #endif
     return buf;
 }

